File: Codility/src/train/time_complexity/TapeEquilibrium.java
package train.time_complexity;

public class TapeEquilibrium {
	public int solution(int[] A) {
		// write your code in Java SE 7
		long a = 0;
		long b = 0;
		a = A[0]; 
		for(int i = 1; i < A.length; i++){
			b += A[i];
		}
		long max = Math.abs(a-b);
		for(int i = 1; i < A.length-1; i++){
			a += A[i];
			b -= A[i];
			long sum = Math.abs(a-b);
			if(sum < max){
				max = sum;
			}
		}
		return (int) max;
	}
	public static void main(String[] args) {
		TapeEquilibrium s = new TapeEquilibrium();
		int[] A = {3000,1000};
		int i = s.solution(A);
		System.out.println(i);
	}

}

File: Codility/src/train/time_complexity/TapeEquilibrium.go
package main

import (
	"fmt"
)

func abs(a int) int {
	if a < 0 {
		return -a
	}
	return a
}

func solution(A []int) int {
		// write your code in Java SE 7
	var a int = 0
	var b  int = 0
	a = A[0]
	for i := 1; i < len(A); i++ {
		b += A[i]
	}
	var max int = abs(a-b)
	for i := 1 ; i < len(A)-1; i++{
		a += A[i]
		b -= A[i]
		var sum int = abs(a-b)
		if(sum < max){
			max = sum
		}
	}
	return max;
}

func main() {
	A := []int{3000,1000}
	fmt.Println("A ==", A)
	i := solution(A)
	fmt.Println(i)
}